backend/controller/ingress/ingress_test.go

```diff
@@ -73,3 +74,28 @@ func TestValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestParseQueryParams(t *testing.T) {
+
+	tests := []struct {
+		query   string
+		request obj
+	}{
+		{query: "", request: obj{}},
+		{query: "a=10", request: obj{"a": "10"}},
+		{query: "a=10&a=11", request: obj{"a": []string{"10", "11"}}},
+		{query: "a.b=10", request: obj{"a": map[string]any{"b": "10"}}},
+		{query: "a.b=10&a.b=11", request: obj{"a": map[string]any{"b": []string{"10", "11"}}}},
+		{query: "a.b.c=10", request: obj{"a": map[string]any{"b": map[string]any{"c": "10"}}}},
+
+		{query: "a.b=10&a=11", request: obj{"a": "11"}}, // last value wins
```

Comment:
This should support `a.0=11` for nested arrays too?

Reply:
Hmm. Maybe I'm missing it, but this is how we currently support nested arrays `a.b=10&a.b=11`. Would the `a.0=11` version allow us the order the array?

Reply:
Oh, do you mean an array in an array vs. an array in a map?

Reply:
Mmm that too!